### PR TITLE
AC-538 Add Snackbar when going online

### PR DESCRIPTION
**Pull request by Abdelaty**
## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-538

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 1]
```diff
@@ -208,7 +210,7 @@ private void showNoInternetConnectionSnackbar() {
         snackbar = Snackbar.make(findViewById(android.R.id.content),
                 getString(R.string.no_internet_connection_message), Snackbar.LENGTH_INDEFINITE);
         View sbView = snackbar.getView();
-        TextView textView = (TextView) sbView.findViewById(com.google.android.material.R.id.snackbar_text);
+        TextView textView = sbView.findViewById(com.google.android.material.R.id.snackbar_text);
```
Comment (Rimjhim28) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: Hey @Abdelaty, this line can be made more readable by importing ```com.google.android.material.R``` and just using ```sbView.findViewById(R.id.snackbar_text);``` out here.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 2]
```diff
@@ -353,4 +355,12 @@ public void onClick(DialogInterface dialogInterface, int i) {
         alertDialog.show();
     }
 
+    private void showConnectedToInternetSnackbar() {
+        snackbar = Snackbar.make(findViewById(android.R.id.content),
+                (getString(R.string.connected_successfully_message)), Snackbar.LENGTH_SHORT);
```
Comment (Rimjhim28) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: This line can be made short by including the required classes in the import statements.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 3]
```diff
@@ -353,4 +355,12 @@ public void onClick(DialogInterface dialogInterface, int i) {
...
+        View sbView = snackbar.getView();
+        TextView textView = sbView.findViewById(com.google.android.material.R.id.snackbar_text);
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: Did you see @Rimjhim28 's comment? Do the same here as well.

**Comment by f4ww4z** [comment]
https://photos.app.goo.gl/pLWvz3iKDvkgUXqJ9

See that the Snackbar is hidden behind the Toast message. I think that showing a Toast message when succesfully connected to Internet is better than adding a Snackbar, since it doesn't stack with other views and it's shown just once every now and then.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java (hunk 1 above)
Comment (Abdelaty) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: Thank you @Rimjhim28  I've changed it.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java (hunk 3 above)
Comment (Abdelaty) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: ok.

**Comment by Abdelaty** [comment]
@f4ww4z so it's better to change the Snackbar to a ToastMessage?

**Comment by f4ww4z** [comment]
> @f4ww4z @Rimjhim28 so it's better to change the Snackbar to a ToastMessage?

Yes please. See my comment on the JIRA issue. @Abdelaty

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 4]
```diff
@@ -205,10 +209,10 @@ public void onNext(List<Location> locations) {
     }
 
     private void showNoInternetConnectionSnackbar() {
-        snackbar = Snackbar.make(findViewById(android.R.id.content),
-                getString(R.string.no_internet_connection_message), Snackbar.LENGTH_INDEFINITE);
+        snackbar = make(findViewById(android.R.id.content),
+                getString(R.string.no_internet_connection_message), LENGTH_INDEFINITE);
```
Comment (deepak140596) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: Revert back to earlier line. Use Snackbar.make and Snackbar.LENGTH_INDEFINITE. See earlier comment as Snackbar was already imported.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 5]
```diff
@@ -58,6 +58,8 @@
 import rx.android.schedulers.AndroidSchedulers;
 import rx.schedulers.Schedulers;
 
+import static com.google.android.material.snackbar.Snackbar.LENGTH_INDEFINITE;
+import static com.google.android.material.snackbar.Snackbar.make;
```
Comment (deepak140596) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: Do not import the snackbar properties. See line 27, Snackbar is already imported.

**Comment by f4ww4z** [comment]
Closing as #499 is the updated PR. @Abdelaty You need to add the link of the updated PR in JIRA.

**Pull request closed without merging**